invenio_rdm_records/marshmallow/json.py

```diff
@@ -36,34 +42,235 @@ class PersonIdsSchemaV1(StrictKeysMixin):
-    recid = PersistentIdentifier()
-    title = SanitizedUnicode(required=True, validate=validate.Length(min=3))
+    additional_descriptions = fields.List(fields.Nested(DescriptionSchemaV1))
+    additional_titles = fields.List(fields.Nested(TitleSchemaV1))
+    contributors = Nested(ContributorSchemaV1, many=True, required=True)
+    dates = fields.List(
+        fields.Nested(DateSchemaV1), validate=validate.Length(min=1))
     description = SanitizedUnicode()
+    embargo_date = DateString()
+    keywords = fields.List(SanitizedUnicode(), many=True)
+    language = SanitizedUnicode()
     owners = fields.List(fields.Integer(),
                          validate=validate.Length(min=1),
                          required=True)
-    keywords = fields.List(SanitizedUnicode(), many=True)
     publication_date = DateString()
-    contributors = Nested(ContributorSchemaV1, many=True, required=True)
+    recid = PersistentIdentifier()
+    resource_type = fields.Nested(ResourceTypeSchemaV1)
+    rights = fields.List(fields.Nested(RightSchemaV1))
+    title = SanitizedUnicode(required=True, validate=validate.Length(min=3))
+    version = SanitizedUnicode()
+
+    @pre_load()
+    def preload_publicationdate(self, data):
+        """Default publication date."""
+        if 'publication_date' not in data:
+            data['publication_date'] = arrow.utcnow().date().isoformat()
+
+    @validates('dates')
+    def validate_dates(self, value):
+        """Validate that start date is before the corresponding end date."""
+        for interval in value:
+            start = arrow.get(interval.get('start'), 'YYYY-MM-DD').date() \
+                if interval.get('start') else None
+            end = arrow.get(interval.get('end'), 'YYYY-MM-DD').date() \
+                if interval.get('end') else None
+
+            if not start and not end:
+                raise ValidationError(
+                    _('There must be at least one date.'),
+                    field_names=['dates']
+                )
+            if start and end and start > end:
+                raise ValidationError(
+                    _('"start" date must be before "end" date.'),
+                    field_names=['dates']
+                )
+
+    @validates('embargo_date')
+    def validate_embargo_date(self, value):
+        """Validate that embargo date is in the future."""
+        if arrow.get(value).date() <= arrow.utcnow().date():
+            raise ValidationError(
+                _('Embargo date must be in the future.'),
+                field_names=['embargo_date']
+            )
+
+    @validates('language')
+    def validate_language(self, value):
+        """Validate that language is ISO 639-3 value."""
+        validate_iso639_3(value)
 
 
-# TODO: Use `RecordMetadataSchemaJSONV1` to inject PID in PUT/PATCH/...
 class RecordSchemaV1(StrictKeysMixin):
     """Record schema."""
 
+    # TODO: Use `RecordMetadataSchemaJSONV1` to inject PID in PUT/PATCH/...
```

Comment:
What does this mean? (maybe just cut / paste in which case, we can get rid of it)

Reply:
I do not think is needed to be tackled now since identifiers are still in progress. However, I implemented [this](https://github.com/inveniosoftware/invenio-records-rest/blob/master/invenio_records_rest/schemas/json.py#L102) because there was a case where:

1- User creates record
2- Somehow through the API the user specified a different `recid` in the body than the one in the URL, or does not specify it in the body.
3- It gets modified/deleted and ends in an inconsistent state.

The fix basically injects the URL pid in the body.

I am also unsure this might happen here since in the end we will always pass by *deposits*, that's why the `TODO`.

Sorry for the lack of context.